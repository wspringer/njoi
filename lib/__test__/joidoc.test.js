const Joi = require("@hapi/joi");

const schema1 = Joi.object()
  .keys({
    a: Joi.string()
      .required()
      .description(
        "Some string alkdsf jlkasj dfjklasj dfklj aklsj fkljaklsdfj klajs fklj aklsj fklja sklfjkl asjdfklj asklj klajsklfj aklsj dfklaskldfj klaj skldf jf"
      )
      .example("yay"),
    b: Joi.number()
      .description("Some number")
      .default(3),
    c: Joi.boolean().description("Some boolean"),
    d: Joi.array().items(Joi.string()),
    e: Joi.array().items(
      Joi.object().keys({
        first: Joi.string()
      })
    )
  })
  .unknown(true);

describe("joidoc", () => {
  const joidoc = require("../joidoc");
  it("should a sensible description", () => {
    console.info(JSON.stringify(schema1.describe(), null, 2));
    expect(joidoc(schema1)).toMatchSnapshot();
  });
});
