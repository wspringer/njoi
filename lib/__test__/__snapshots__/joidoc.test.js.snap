// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joidoc should a sensible description 1`] = `
"{
  /**
   * Some string alkdsf jlkasj dfjklasj dfklj aklsj 
   * fkljaklsdfj klajs fklj aklsj fklja sklfjkl 
   * asjdfklj asklj klajsklfj aklsj dfklaskldfj klaj 
   * skldf jf
   */
  a: \\"yay\\",

  /**
   * Some number
   */
  b: <number>,

  /**
   * Some boolean
   */
  c: <boolean>,

  d: [<string>, ...],

  e: [{
    first: <string>,
  }, ...],

  type: <string>,
}"
`;
